src/cli/commands/retrieve_metrics.go

```diff
@@ -133,3 +148,67 @@ func RetrieveMetrics(cliConnection api.Connection, appName, metricName string, s
 
 	return nil
 }
+
+func RetrieveAggregatedMetrics(cliConnection api.Connection, appName, metricName string, startTime, endTime int64, desc bool, writer io.Writer) error {
+
+	cfclient, err := api.NewCFClient(cliConnection)
+	if err != nil {
+		return err
+	}
+	err = cfclient.Configure(appName)
+	if err != nil {
+		return err
+	}
+
+	endpoint, err := api.GetEndpoint()
+	if err != nil {
+		return err
+	}
+	if endpoint.URL == "" {
+		return errors.New(ui.NoEndpoint)
+	}
+
+	apihelper := api.NewAPIHelper(endpoint, cfclient, os.Getenv("CF_TRACE"))
+	ui.SayMessage(ui.ShowAggregatedMetricsHint, appName)
+
+	table := ui.NewTable(writer, []string{"Metrics Name", "Value", "Timestamp"})
+	var (
+		page     uint64 = 1
+		next     bool   = true
+		noResult bool   = true
+		data     [][]string
+	)
+	for next {
+		next, data, err = apihelper.GetAggregatedMetrics(metricName, startTime, endTime, desc, page)
+		if err != nil {
+			return err
+		}
+
+		for _, row := range data {
+			table.Add(row)
+		}
+		if len(data) > 0 {
+			noResult = false
+			table.Print()
+		}
+
+		if !next {
+			break
+		}
+		page += 1
+	}
+
+	if noResult {
+		ui.SayOK()
+		ui.SayMessage(ui.AggregatedMetricsNotFound, appName)
+		ui.SayMessage(ui.UseInstanceMetricMsg)
+
+	} else {
+
+		if writer != os.Stdout {
+			ui.SayOK()
```

Comment:
why?

Reply:
only print OK when the data is printed to a file.  Like this:
```
cf asm nodetestapp1 memoryused --start 2018-09-11T12:59:13+08:00 --end 2018-09-11T13:04:33+08:00 --output output.txt
Showing aggregated metrics for app nodetestapp1...
OK
```

Reply:
this is weird ...  

1) the progressing message should be "retrieving aggregated metrics of app nodetestapp1..."
2) "OK" basically mean you successfully retrieved the metrics 
3) you probably will need to show metrics saved in file "xxx"

Reply:
This will be addressed in a New PR